freqtrade/strategy/interface.py

```diff
@@ -241,8 +242,26 @@ def _analyze_ticker_internal(self, dataframe: DataFrame, metadata: dict) -> Data
 
         return dataframe
 
-    def get_signal(self, pair: str, interval: str,
-                   dataframe: DataFrame) -> Tuple[bool, bool]:
+    @staticmethod
+    def preserve_df(dataframe: DataFrame) -> Tuple[int, float, datetime]:
+        """ keep some data for dataframes """
+        return len(dataframe), dataframe["close"].iloc[-1], dataframe["date"].iloc[-1]
+
+    @staticmethod
+    def assert_df(dataframe: DataFrame, df_len: int, df_close: float, df_date: datetime):
+        """ make sure data is unmodified """
+        message = ""
+        if df_len != len(dataframe):
+            message = "length"
+        elif df_close != dataframe["close"].iloc[-1]:
+            message = "last close price"
+        elif df_date != dataframe["date"].iloc[-1]:
+            message = "last date"
+        if message:
+            raise DependencyException("Dataframe returned from strategy has mismatching "
+                                      f"{message}.")
```

Comment:
I'm not certain about the type of the exception yet - `Exception` (what you had before) was definitely wrong ... 

but we'll need to ask the question - what do we want to archive?

should the bot be stopped completely since the strategy is wrong? 
or should just this iteration / cycle be interrupted and retried next time (assuming it's a general error - i guess that would lead to the bot being stuck / seeming stuck).

If we want it stopped, we can remove / skip this check after it ran once saving processing time - assuming the strategy won't act differently after the first iteration.
If on the other hand we assume a temporary error (not sure that's even possible without intentionally doing that) - then we'll need to run it each time - but can't crash the bot completely, but need to skip this iteration.

Reply:
If it catches the error on the first iteration it's probably a configuration error, so it stops right away.
If it appears after a while, the strategy is relieing on some currently unavailable data, should keep trying.